### PR TITLE
Checking Ubuntu compatibility by parsing collections yaml

### LINKED ISSUE #49: Parse collections from YAML to get Ubuntu compatibility
## Summary

The current implementation uses an internal object to hold the compatible Gazebo/Ubuntu combinations. Instead of this, it would be effective to parse the collections YAML to get the corresponding Ubuntu distributions.

## Implementation suggestion
This collection file can be parsed to find check the compatible ubuntu distribution 
https://raw.githubusercontent.com/gazebo-tooling/release-tools/master/jenkins-scripts/dsl/gz-collections.yaml

### PR DESCRIPTION
**Pull request by sauk2**
Closes #49 

This PR parses a collection yaml to check ubuntu compatibility, replacing the operation of reading an internal object.

### CODE REVIEW

**Comment by sauk2** [comment]
One point I missed mentioning, I am still checking if it is a valid gazebo distro using [this](https://github.com/sauk2/setup-gazebo/blob/3459f6b0292d3ed7030dfe07634d5c3a13c742a7/src/utils.ts#L8-L32) object

```node
const validGazeboDistroList: {
	name: string;
	compatibleUbuntuDistros: string[];
}[] = [
	{
		name: "citadel",
		compatibleUbuntuDistros: ["focal"],
	},
	{
		name: "fortress",
		compatibleUbuntuDistros: ["focal", "jammy"],
	},
	{
		name: "garden",
		compatibleUbuntuDistros: ["focal", "jammy"],
	},
	{
		name: "harmonic",
		compatibleUbuntuDistros: ["jammy", "noble"],
	},
	{
		name: "ionic",
		compatibleUbuntuDistros: ["noble"],
	},
];
```

Do you think it is a good idea to use the collections to get the list of valid gazebo distributions? Or do you think it would be better to simplify the object to carry a list of valid distributions?

File: src/utils.ts [hunk 1]
```diff
@@ -125,23 +126,33 @@ export function getRequiredGazeboDistributions(): string[] {
...
+		"https://raw.githubusercontent.com/gazebo-tooling/release-tools/master/jenkins-scripts/dsl/gz-collections.yaml",
+	)
+		.then((response) => response.blob())
+		.then((blob) => blob.text())
+		.then((yamlStr) => {
+			const collections = parseDocument(yamlStr).toJSON();
+			collections["collections"].forEach((obj: any) => {
+				requiredGazeboDistributionsList.forEach((element) => {
+					if (obj["name"] === element) {
+						if (obj["packaging"]["configs"].indexOf(ubuntuCodename) <= -1) {
```
Comment (j-rivero) [inline src/utils.ts:145]: We need to be careful here. The packaging.configs is a configuration name that most of the times matches the Ubuntu distro name but NOT always. See https://github.com/gazebo-tooling/release-tools/blob/master/jenkins-scripts/dsl/gz-collections.yaml#L346 noble_no_pr . With that name we need to check in pakaging_configs section in the yaml the system.so.distribution == ubuntu and system.so.version == ubuntuCodename.

**Review by j-rivero — CHANGES_REQUESTED** [review]
Important change in the algorithm

**Comment by j-rivero** [comment]
I've added some comments and rename some variables in this commit https://github.com/gazebo-tooling/setup-gazebo/commit/cd398c92448e30cd24380265b66e9a1f0ad59174 . Sorry that my dev environment is kind of broken now and I was not able to commit it properly to the project. Could you please get the comments and the var renames into this PR?

**Review by j-rivero — COMMENTED** [review]
The PR is good to go I would say. The great point of making the change to use the yaml file to finally remove the harcoded list of Gazebo distros mapping in https://github.com/gazebo-tooling/setup-gazebo/blob/main/src/utils.ts#L7-L31. There are more changes to do in order to so so but we can use a new PR for that.

**Comment by sauk2** [comment]
I have implemented all the requested changes. All checks pass!

**Comment by j-rivero** [comment]
> I have implemented all the requested changes. All checks pass!

Thanks so much ! Great work!

**Pull request merged**